src/protocols/user_directed_commissioning/UserDirectedCommissioningClient.cpp

```diff
@@ -32,8 +32,30 @@ namespace UserDirectedCommissioning {
 CHIP_ERROR UserDirectedCommissioningClient::SendUDCMessage(TransportMgrBase * transportMgr, System::PacketBufferHandle && payload,
                                                            chip::Transport::PeerAddress peerAddress)
 {
-    CHIP_ERROR err;
+    CHIP_ERROR err = EncodeUDCMessage(std::move(payload));
+    if (err != CHIP_NO_ERROR)
+    {
+        return err;
+    }
+    ChipLogProgress(Inet, "Sending UDC msg");
+
+    // send UDC message 5 times per spec (no ACK on this message)
+    for (unsigned int i = 0; i < 5; i++)
+    {
+        err = transportMgr->SendMessage(peerAddress, std::move(payload));
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "UDC SendMessage failed, err: %s\n", chip::ErrorStr(err));
+            return err;
+        }
+        sleep(1);
```

Comment:
So this function will block the caller's thread for <= 5 seconds. It may not be an issue now since the function is called only from the shell, but in real-life we would probably need to use some timers to be able to run this function on the main CHIP thread, correct? Do you have any plans with regards to that?

Reply:
This method is expected to be called interactively (upon user interaction) - see 5.2.2.4 "Launch Device User Interface, Discover Commissioners".